processor/src/main/java/org/mapstruct/ap/internal/model/source/BeanMappingOptions.java

```diff
@@ -55,15 +55,26 @@ public static BeanMappingOptions forInheritance(BeanMappingOptions beanMapping,
         return options;
     }
 
-    public static BeanMappingOptions forForgedMethods(BeanMappingOptions beanMapping) {
-        BeanMappingOptions options = new BeanMappingOptions(
+    public static BeanMappingOptions forForgedMethods(MappingMethodOptions mappingMethodOptions) {
+
+        BeanMappingOptions beanMapping = mappingMethodOptions.getBeanMapping();
+
+        return new BeanMappingOptions(
             beanMapping.selectionParameters != null ?
                 SelectionParameters.withoutResultType( beanMapping.selectionParameters ) : null,
-            Collections.emptyList(),
+            takeOverIgnoreUnmappedSourceProperties( mappingMethodOptions ),
             beanMapping.beanMapping,
             beanMapping
         );
-        return options;
+    }
+
+    private static List<String> takeOverIgnoreUnmappedSourceProperties(MappingMethodOptions mappingMethodOptions) {
+
+        if (!mappingMethodOptions.getSubclassMappings().isEmpty()) {
```

Comment:
...so that it is now only used when dealing with subclass mappings. Might argue for possibly refining that.